### DISCUSSION TITLE
Inline Debugging with Platformio and VS Code

### DISCUSSION THREAD

**Discussion by jep342** [hatlabs/discussions#15] [Q&A]
Great Board!!  I took delivery of the board a few days ago and have it up and running without any problems. Now, i want to do some customization of software.  Since i am new to the platformio and esp32 environment, i have a question about inline debugging and was not sure if this was supported on the SH-ESP32 directly or whether I would need to use something like an ESP-prog board?

I presume the alternative would be to output some development debug messages in the terminal window to verify proper function of my development code.

Do you have a recommended approach for debugging the SH-ESP32?

Jim

**Comment by jep342** [comment]
Apologies for the newbie channel clutter…  I found the discussion in the documentation regarding the JTAG connections on the SH-ESP32 board.  This answers my question. 
Jim

**Reply by mairas** [reply]
Hi, Jim!

No worries! Yes, the JTAG connection is indeed provided on the board. I have to admit, though, that I've never used it myself. I do have the necessary hardware ... somewhere, but good old "printf" debugging has served me quite well. But if you decide to have a go with JTAG, please do share your experiences!
